File: DSA/DSA08007 SỐ BDN 1.cpp
#include <bits/stdc++.h>
using namespace std;

int to_int(string s)
{
    int n = 0;
    for (auto &&i : s)
    {
        n = n*2;
        if (i == '1')
        {
            n++;
        }
    }
    return n;
}

void submit()
{
    string s;
    cin >> s;
    bool check = false;

    for (auto &&i : s)
    {
        if (i - '0' > 1 || check)
        {
            i = '1';
            check = true;
        }
    }
    
    cout << to_int(s) << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed down input/output
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    
    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}

File: DSA_python/DSA08007 SỐ BDN 1.py
def submit():
    s = input()
    A = list(s)
    check = False

    for i in range(len(s)):
        if ord(s[i]) - ord('0') > 1 or check == True:
            A[i] = "1"
            check = True
    
    def to_int(A):
        n = 0
        for i in A:
            n = n*2
            if i == '1':
                n+=1
        return n
    
    print(to_int(A))





for i in range(int(input())):
    submit()
